make None the fall back return value of infer_type

bernt-matthias opened the pull request:
all functions using infer_type already return CliString as fallback
but they currently check for None

bernt-matthias committed:
make None the fall back return value

all functions using infer_type already return CliString as fallback
but they currently check for None

multimeric commented:
This is a good change, but it requires adding an `if None` check to various functions that use `infer_type`, e.g. `SimpleFlagArg.get_type()`. That's why the test is failing.

bernt-matthias committed:
add default parameter for infer_type

multimeric commented:
Looks good! But actually I think the more idiomatic Python would be to have it return `None` and then use `or`, e.g. 
```python
infer_type(self.description, default=cli_types.CliString())
```
becomes
```python
infer_type(self.description) or cli_types.CliString()
```

bernt-matthias committed:
make it more idiomatic python

default return of infer_type is again None and

infer_type(name, default=Type) becomes infer_type(name) or Type

multimeric commented:
Excellent, thanks for the work on this PR!
